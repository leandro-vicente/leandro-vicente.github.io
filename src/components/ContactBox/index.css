form {
  width: 100%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  padding-top: 20px;
}

.form-container {
  width: 100%;
}

.btn-action {
  margin-left: 20px;
  margin-right: 20px;
}

