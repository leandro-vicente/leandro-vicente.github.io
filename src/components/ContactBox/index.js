import React, { Component } from 'react'
import axios from 'axios'
import './index.css'

const GOOGLE_FORM_MESSAGE_ID = 'entry.1249636003'
const GOOGLE_FORM_EMAIL_ID = 'entry.1157005188'
const GOOGLE_FORM_ACTION = 'https://docs.google.com/forms/d/e/1FAIpQLScJX66KS2GUD1Ci-piKoPbsracqXFjJr7kSWd3yAin_vgk7OQ/formResponse'

class ContactBox extends Component {
  constructor(props){
    super(props)
    this.state = { 
      message: '', 
      email: '', 
      showForm: false,
      sendingMessage: false,
      messageSent: false,
      messageError: false
    }
  }

  handleChange = (event) => {
    this.setState({[event.target.name]: event.target.value})
  }

  handleSubmit = (event) => {
    event.preventDefault()
    this.setState({
      sendingMessage: true
    })
    this.sendMessage()
  }

  handleFormToggle = () => {
    this.setState( 
      (prevState) => {
        const { showForm } = prevState
        return {
          showForm: !showForm
        } 
      }
    )
  }

  handleReturnButton = () => {
    this.setState(
      {
        showForm: false,
        messageSent: false,
        messageError: false
      }
    )
  }

  sendMessage = () => {
    const formData = new FormData()
    formData.append(GOOGLE_FORM_MESSAGE_ID, this.state.message)
    formData.append(GOOGLE_FORM_EMAIL_ID, this.state.email)

    axios.post(GOOGLE_FORM_ACTION,formData)
      .then(() => {
        this.setState({ 
          messageSent: true,
          sendingMessage: false,
          message: '',
          email: ''
        })
      }).catch(() => {
        this.setState({ 
          messageError: true,
          sendingMessage: false
        })
      })
  }

  returnButton = () => <button id='return-button' className='btn btn-default btn-xs' onClick={this.handleReturnButton}>Return</button>

  render() {
    if(this.state.sendingMessage){
      return(
        <div>Sending...</div>
      )
    }

    if(this.state.messageSent){
      return(
        <React.Fragment>
          <div className='success-message'>Sent! I will respond asap</div>
          {this.returnButton()}
        </React.Fragment>
      )
    }

    if(this.state.messageError){
      return(
        <React.Fragment>
          <div className='error-message'>Sorry, your message was not sent. Contact me at: leandrovicente.info@gmail.com</div>
          {this.returnButton()}
        </React.Fragment>
      )
    }

    if(this.state.showForm === false){
      return(
        <button id='contact-button' className='btn btn-sm' onClick={this.handleFormToggle}>Contact</button>
      )
    }

    return(
      <React.Fragment>
        <div className='form-container'>
          <form onSubmit={this.handleSubmit}>
            <div className='form-group row'>
              <label htmlFor='email' className='col-sm-2 col-form-label'>
                Email:
              </label>
              <div className='col-sm-8'>
                <input 
                  type='email' 
                  name='email'
                  id='email'
                  className='form-control'
                  value={this.state.email} 
                  onChange={this.handleChange}
                  required
                />
              </div>
            </div>
            <div className='form-group row'>
              <label htmlFor='message' className='col-sm-2 col-form-label'>
                Message:
              </label>
              <div className='col-sm-8'>
                <textarea 
                  id='message'
                  name='message'
                  className='form-control'
                  required
                  value={this.state.message} 
                  onChange={this.handleChange} 
                  rows='6'
                />
              </div>
            </div>
            <div>
              <button type='button' id='cancel-button' className='btn btn-default btn-sm btn-action' onClick={this.handleFormToggle}>Cancel</button>
              <button type='submit' className='btn btn-sm btn-default btn-action'>Submit</button>
            </div>
          </form>
        </div>
      </React.Fragment>
    )
  }
}

export default ContactBox