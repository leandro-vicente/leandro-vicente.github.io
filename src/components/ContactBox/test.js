import React from 'react'
import { shallow } from 'enzyme'
import ContactBox from '.'

describe('Contact Box', () => {
  let contactBox

  beforeEach(() => {
    contactBox = shallow(<ContactBox />)
  })

  it('Renders correctly', () => {
    expect(contactBox).toMatchSnapshot()
  })

  it('renders with the form hidden by default', () => {
    expect(contactBox.find('form').length).toBe(0)
  })

  it('renders with the contact button by default', () => {
    expect(contactBox.find('#contact-button').length).toBe(1)
  })

  it('shows form when the contact button is clicked ', () => {
    contactBox.find('#contact-button').simulate('click')

    expect(contactBox.find('form').length).toBe(1)
  })

  it('hides form when the cancel button is clicked', () => {
    contactBox.setState({showForm: true})
    contactBox.find('#cancel-button').simulate('click')

    expect(contactBox.find('form').length).toBe(0)
  })

  it('shows contact button when the cancel button is clicked', () => {
    contactBox.setState({showForm: true})
    contactBox.find('#cancel-button').simulate('click')

    expect(contactBox.find('#contact-button').length).toBe(1)
  })
})