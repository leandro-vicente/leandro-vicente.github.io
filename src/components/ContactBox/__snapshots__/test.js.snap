// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact Box Renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ContactBox />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Contact",
      "className": "btn btn-sm",
      "id": "contact-button",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": "Contact",
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Contact",
        "className": "btn btn-sm",
        "id": "contact-button",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": "Contact",
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
